### DISCUSSION TITLE
[Competing risks] Does default `cause` setting always generating lower OOB errors than cause-specific settings?

### DISCUSSION THREAD

**Discussion by ringo-alin** [kogalur/randomForestSRC#205] [Q&A]
Hi, I did some tests and found this counterintuitive results. I thought that at least the cause-specific OOB error of the latter setting should be lower.  Is it the model's expected behavior?
```
library(randomForestSRC)
library(survival)
set.seed(123)
data(pbc, package = "survival")
pbc$id <- NULL
pbcn <- impute(Surv(time, status)~., pbc)
o1 <- rfsrc(Surv(time, status)~., pbcn)
print(o1$err.rate[500,])
# event.1   event.2 
# 0.2251481 0.1512610
o1 <- rfsrc(Surv(time, status)~., pbcn, cause = c(1, 0))
print(o1$err.rate[500,])
# event.1   event.2 
# 0.2488479 0.1639438 
o1 <- rfsrc(Surv(time, status)~., pbcn, cause = c(0, 1))
print(o1$err.rate[500,])
# event.1   event.2 
# 0.2999781 0.1586431 
```

**Comment by ishwaran** [comment]
Your three calls are fitting three different models.  The first call is the default setting which uses the default splitting rule a weighted Gray's test.  The second call is also using the default splitting rule, but where the weights are 1 for cause 1 and 0 for cause 2.  The third call is like the second call but now the weights are 0 for cause 1 and 0 for cause 2.

Therefore since you are fitting three different types of models, the results will be different.
